### ISSUE TITLE
SSO: Assume Role not working from source_profile of SSO

### ISSUE THREAD

**Issue by mil1i** [aws/aws-toolkit-vscode#1740]
**Describe the bug**
I have a profile that connects via sso_*, chooses an SSO role. Then a second profile that uses that defines "role_arn", and the "source_profile" pointed to the SSO profile.

This assume role profile works in Jetbrains AWS Toolkit, Visual Studio AWS Toolkit and also when specifying "--profile blah" on awscli-v2.

When attempting to utilize this profile in Visual Studio Code, it produces an error.

**To Reproduce**

```
[profile sso]
sso_start_url = https://${SSO_START_URL}.awsapps.com/start
sso_region = us-east-1
sso_account_id = ${AWS_ACCOUNT}
sso_role_name = ${SSO_ROLE_NAME}
region = us-east-1

[profile assume-role]
source_profile = sso
role_arn = arn:aws:iam::${AWS_ACCOUNT}:role/my-role
```

**Expected behavior**
Establish connection/authorization utilizing the profile to assume the role.

**Screenshots**

```less
2021-05-06 21:28:17 [ERROR]: log level: info
2021-05-06 21:28:17 [INFO]: Retrieving AWS endpoint data
2021-05-06 21:28:17 [INFO]: OS:  Darwin x64 20.4.0
2021-05-06 21:28:17 [INFO]: Visual Studio Code Extension Host Version:  1.56.0
2021-05-06 21:28:17 [INFO]: AWS Toolkit Version:  1.24.0
2021-05-06 21:28:17 [INFO]: telemetry cache not found: '/Users/user/Library/Application Support/Code/User/globalStorage/amazonwebservices.aws-toolkit-vscode/telemetryCache'
2021-05-06 21:28:37 [ERROR]: Error trying to connect to AWS with Credentials Provider profile:assume-role. Toolkit will now disconnect from AWS. [Error [CredentialsError]: Missing credentials in config, if using AWS_CONFIG_FILE, set AWS_SDK_LOAD_CONFIG=1
	at constructor.load (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1649:6028)
	at constructor.coalesceRefresh (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1438:26701)
	at constructor.refresh (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1649:6223)
	at constructor.get (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1438:26529)
	at f (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1638:27945)
	at constructor.getCredentials (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1638:28226)
	at constructor.<anonymous> (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1638:30721)
	at constructor.callListeners (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1438:24989)
	at constructor.emit (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1438:24739)
	at constructor.emitEvent (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1639:10845)
	at constructor.N (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1639:6431)
	at u.runTo (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1639:12683)
	at constructor.runTo (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1639:8122)
	at constructor.send (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1639:8015)
	at r.constructor.r.constructor.makeRequest (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1638:17058)
	at r.constructor.r.constructor.e.authtype.d.<computed> [as assumeRole] (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1638:23928)
	at constructor.loadRoleProfile (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1649:7497)
	at constructor.load (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1649:5604)
	at constructor.coalesceRefresh (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1438:26701)
	at constructor.refresh (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1649:6223)
	at constructor.get (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1438:26529)
	at new constructor (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1649:5199)
	at /Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:2204:3416
	at f (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1438:27709)
	at constructor.resolve (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1438:27806)
	at /Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:364:15558
	at new Promise (<anonymous>)
	at constructor.resolvePromise (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:364:15498)
	at bS.getCredentials (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:2204:1797)
	at zS.setCredentials (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:2625:587)
	at zS.upsertCredentials (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:2625:449)
	at async ef.login (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:2216:327)
	at async Ik.onCommandLogin (/Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:2526:310)
	at async /Users/user/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:2940:1140
	at async E._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:87:107924)] {
  code: 'CredentialsError',
  time: 2021-05-07T03:28:37.728Z,
  originalError: {
    message: 'Could not load credentials from constructor',
    code: 'CredentialsError',
    time: 2021-05-07T03:28:37.728Z,
    originalError: {
      message: 'Credentials not set for profile sso',
      code: 'SharedIniFileCredentialsProviderFailure',
      time: 2021-05-07T03:28:37.728Z
    }
  }
}
```

**Desktop (please complete the following information):**

-   OS: macOS 11.3.1 && Windows 10
-   Visual Studio Code Version: 1.56.0
-   AWS Toolkit Version: 1.24.0

**Additional context**

Profile works as expected in Jetbrains and Visual Studio AWS Toolkit, as well with awscli-v2.

**Comment by JadenSimon** [comment]
This is related to #1655. Should be fixed when we finish migrating to v3 of the SDK.

**Comment by JadenSimon** [comment]
@mil1i 
We've been working on a fix for this issue. I've tested SSO credentials with role assumption and it appears to be working.

To confirm the fix works for your use case, it would be helpful you can try out the pre-release build:
1. Download and extract [buildArtifacts.zip](https://github.com/aws/aws-toolkit-vscode/files/6540621/buildArtifacts.zip)
2. From the VSCode command list (ctrl-shift-p or cmd-shift-p) run Extensions: Install from VSIX... to install the *.vsix file.
3. Reload VSCode and verify that the Toolkit has version 1.26 (not the release version 1.25).

**Comment by mil1i** [comment]
It does appear to be working with v1.26 build.

![image](https://user-images.githubusercontent.com/8459426/119692756-e7a07300-be08-11eb-973e-3106dbfd941c.png)

![image](https://user-images.githubusercontent.com/8459426/119692471-ae680300-be08-11eb-9fa6-45a834210759.png)

**Comment by justinmk3** [comment]
Fixed in AWS Toolkit 1.29
